### PR TITLE
[TensorRT EP] Refactor OrtTensorRTProviderOptions initialization and make it easy to add new field

### PR DESCRIPTION
**Pull request by chilo-ms**
Two major modifications of this PR:

1. Refactor OrtTensorRTProviderOptions initialization and make it easy to add new field.
2. Make Python API capable of using TensorRT plugins by adding new Python binding api `register_tensorrt_plugins_as_custom_ops`. (It needs to register ep's custom op domain before model load. For C++ API, it's slightly different, when calling SessionOptionsAppendExecutionProvider_TensorRT_XX, it appends cutom op domain to session option. Later ORT can register custom op domain from session option before model loading)

### CODE REVIEW

File: include/onnxruntime/core/providers/tensorrt/tensorrt_provider_factory.h [hunk 1]
```diff
@@ -1,14 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-#include "onnxruntime_c_api.h"
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_Tensorrt, _In_ OrtSessionOptions* options, int device_id);
```
Comment (jywu-msft) [inline include/onnxruntime/core/providers/tensorrt/tensorrt_provider_factory.h:10]: thanks. been wanting to remove this file.

**Comment by jywu-msft** [comment]
i kicked off python packaging pipeline to test this PR

**Comment by jywu-msft** [comment]
fix python lint checks

**Review by jywu-msft — DISMISSED** [review]

**Comment by jywu-msft** [comment]
are there any code examples , inference-examples that need to be updated now that tensorrt_provider_factory.h is removed?

**Review by jywu-msft — APPROVED** [review]

**Comment by gcunhase** [comment]
Is `register_tensorrt_plugins_as_custom_ops` also available in the Python API? @chilo-ms @jywu-msft

**Pull request merged**